src/coreclr/src/jit/valuenum.cpp

```diff
@@ -4332,6 +4337,51 @@ bool ValueNumStore::IsVNUnsignedCompareCheckedBound(ValueNum vn, UnsignedCompare
     return false;
 }
 
+//------------------------------------------------------------------------
+// IsVNUnsignedCompareConstCheckBound: Checks if the specified vn represents an expression
+//    such as "(uint)cnst <= (uint)len" that implies constant indexes less then cnst are valid
+//
+// Arguments:
+//    vn - Value number to query
+//    info - Pointer to an UnsignedCompareCheckedBoundInfo object to return information about
```

Comment:
> Should it be required only for ** where it makes sense to init it as *outParam = nullptr, or should it be used here as well?
I think that was the intention when the coding conventions were written. The documentation here should be very clear about the contract for who initializes the `*info` value.

Reply:
Additionally since info can never be nullptr, it makes more sense to pass it by reference:

`bool ValueNumStore::IsVNUnsignedCompareConstCheckBound(ValueNum vn, UnsignedCompareCheckedBoundInfo& info)`

That way we know that it was already initialized by the caller.

Reply:
> Additionally since info can never be nullptr, it makes more sense to pass it by reference:

coding conv forbids passing non-const references.